### PR TITLE
Update reclaim-ai extension

### PR DESCRIPTION
**Pull request by shkreios**
## Description

This pull request introduces a utility function `filterMultipleOutDuplicateEvents`. The function filters out events that are synchronized, managed by Reclaim, and part of multiple calendars. To achieve this, the function groups events based on their start and end times, then filters out the irrelevant events based on specific criteria (e.g., event title matching `"Personal Commitment"`, `"Work Commitment"`, or `"Travel Commitment"`, and events that are both synchronized and managed by Reclaim).

Additionally, the `filterMultipleOutDuplicateEvents` function has been integrated into the `useEvent` hook and the `notifications` component to ensure that the displayed events are filtered according to the new logic.

Here’s a detailed summary of changes made in this PR:
- Added `filterMultipleOutDuplicateEvents` utility function to `src/utils/events.ts`.
- Updated `useEvent` hook in `src/hooks/useEvent.tsx` to use `filterMultipleOutDuplicateEvents` for filtering events.
- Updated `notifications` component in `src/notifications.tsx` to use `filterMultipleOutDuplicateEvents` for filtering events.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Thanks for another update @shkreios 🔥 Do you mind adding a [CHANGELOG](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history) entry?

@ajohnson1200 do you mind checking this 🙂

File: extensions/reclaim-ai/src/utils/events.ts [hunk 1]
```diff
@@ -23,3 +23,58 @@ export const truncateEventSize = (eventTitle: string) => {
...
+
+/**
+ * Filter events based on the given criteria.
+ * @param events List of events to filter.
+ * @returns Filtered list of events.
+ */
+const filterRelevantEvents = (events: ApiResponseEvents) => {
+  return events.filter((event) => {
+    const isSyncedAndManaged = event.personalSync && event.reclaimManaged;
+    const matchesTitleCriteria =
```
Comment (ajohnson1200) [inline extensions/reclaim-ai/src/utils/events.ts:51]: Talked with @lightbody about this one, we have an internal (shared here) doc:

https://docs.google.com/document/d/1Hc2d1BE9-Er6-UUoD6v7eZRAJGCigiVHOcotnSuZgCs/edit?usp=sharing

that details, I think, how you could very very reliably detect / link sync events to the source event, without needing to look at the title and the start / end time. Maybe take that doc for a spin and let me know if you think that would simplify things?

Comment (pernielsentikaer) [reply extensions/reclaim-ai/src/utils/events.ts:51]: Thanks for the information! It might indeed be a better way @shkreios 🙂

Comment (shkreios) [reply extensions/reclaim-ai/src/utils/events.ts]: @ajohnson1200, I greatly appreciate your input. I've refined the filtering logic based on your suggestions. However, I'm not entirely confident that my updated implementation accurately addresses the corner cases of recurring events. Could you, or perhaps @lightbody, spare a moment to review it? Thank you!

**Review by ajohnson1200 — APPROVED** [review]
LGTM!

**Pull request merged**